azure-mgmt-trafficmanager/src/main/java/com/microsoft/azure/management/trafficmanager/TrafficManagerEndpoint.java

```diff
@@ -0,0 +1,500 @@
+ * license information.
+ */
+package com.microsoft.azure.management.trafficmanager;
+
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.ExternalChildResource;
+import com.microsoft.azure.management.resources.fluentcore.model.Attachable;
+import com.microsoft.azure.management.resources.fluentcore.model.Settable;
+import com.microsoft.azure.management.resources.fluentcore.model.Wrapper;
+import com.microsoft.azure.management.trafficmanager.implementation.EndpointInner;
+
+/**
+ * An immutable client-side representation of an Azure traffic manager profile endpoint.
+ */
+@Fluent
+public interface TrafficManagerEndpoint extends
+        ExternalChildResource<TrafficManagerEndpoint, TrafficManagerProfile>,
+        Wrapper<EndpointInner> {
+    /**
+     * @return the endpoint type
+     */
+    EndpointType endpointType();
+
+    /**
+     * @return the monitor status of the endpoint
+     */
+    EndpointMonitorStatus monitorStatus();
+
+    /**
+     * @return true if the endpoint is disabled, false otherwise
+     */
+    boolean isDisabled();
+
+    /**
+     * @return the weight of the endpoint which is used when traffic manager profile is configured with
+     * Weighted traffic-routing method
+     */
+    int routingWeight();
+
+    /**
+     * @return the priority of the endpoint which is used when traffic manager profile is configured with
+     * Priority traffic-routing method
+     */
+    int routingPriority();
+
+    /**
+     * The entirety of a traffic manager profile endpoint definition as a part of parent definition.
+     *
+     * @param <ParentT> the return type of the final {@link Attachable#attach()}
+     */
+    interface Definition<ParentT> extends
+            DefinitionStages.Blank<ParentT>,
+            DefinitionStages.WithEndpointType<ParentT>,
+            DefinitionStages.WithEndpointThreshold<ParentT>,
+            DefinitionStages.WithSourceTrafficLocation<ParentT>,
+            DefinitionStages.WithAttach<ParentT> {
+    }
+
+    /**
+     * Grouping of traffic manager profile endpoint definition stages as a part of parent traffic manager
+     * profile definition.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of a traffic manager profile endpoint definition.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface Blank<ParentT> extends WithEndpointType<ParentT> {
+        }
+
+        /**
+         * The stage of the traffic manager profile endpoint definition allowing to specify the type.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithEndpointType<ParentT> {
+            /**
+             * Specify that the endpoint is an Azure cloud service, web app or public ip.
```

Comment:
>Specify that the endpoint is an Azure cloud service, web app or public ip. [](start = 15, length = 74)

not sure I understand this comment  - the API is asking for a resource Id, but the comment suggests it would be a boolean. If the goal is fr the user to specify the resource Id or some resource, then the comment should explain that.  Should it be something like "Specifies the resource ID of a ..."?